### PR TITLE
init: init aws-cdk

### PR DESCRIPTION
**Pull request by rlexus90**
I use aws-cdk
s3 bucket link http://asw-shop.s3-website.eu-west-2.amazonaws.com/
cloudFront link https://d28bx8wya4z9xy.cloudfront.net/

score 100/100

### CODE REVIEW

**Review by VoliaSi — COMMENTED** [review]
Haven't found automated invalidation. Could you point me please, as I'm not good in js and ts. So far - left without a mark. Looking forward to a response.

**Comment by rlexus90** [comment]
There is no separate code for invalidation. It is not broken, when it is deployed, a invalidation
 is automatically created.

Знімок екрана 2024-06-11 о 12 53 18

**Comment by VoliaSi** [comment]
Hi.
I haven't finished this task, so I can't say for sure, but the tasks says:
[image: image.png]

And it looks like you need to have some script or plugin which will
invalidate periodically...
There is a possibility to raise a discussion. I can if you want. Or you can
try to explain to me yourselft...
What do you prefer?
But I need to give you some score quite quickly and I do not want it to be
lower than your work is worth...
Best,
Volha.
З павагай,
Вольга.


вт, 11 июн. 2024 г. в 13:56, rlexus90 ***@***.***>:

**Comment by VoliaSi** [comment]
Anton Kustikov — Сегодня, в 17:56
Invalidation should be triggered, once content of the bucket is updated.
Since content refer to the bucket deployment - we we may take a look at the
documentation.
https://docs.aws.amazon.com/cdk/api/v2/python/aws_cdk.aws_s3_deployment/BucketDeployment.html
There is a parameter in that class, which serves excatly that purpose.
distribution (Optional[IDistribution]) – The CloudFront distribution using
the destination bucket as an origin. Files in the distribution’s edge
caches will be invalidated after files are uploaded to the destination
bucket. Default: - No invalidation occurs distribution_paths (Optional
[Sequence[str]]) – The file paths to invalidate in the CloudFront
distribution. Default: - All files under the destination bucket key prefix
will be invalidated.
З павагай,
Вольга.


вт, 11 июн. 2024 г. в 17:20, Volha Sidoryk ***@***.***>:

**Comment by rlexus90** [comment]
This code is responsible for automatic Invalidation:
    new deploy.BucketDeployment(this, 'deploy', {
      sources: [deploy.Source.asset('../dist')],
      destinationBucket: bucket,
      **distribution: distribution,**
      distributionPaths: ['/*'],
    });

**Comment by VoliaSi** [comment]
THank you for your clarification.
I've discovered that and I've scored you with 100 - you can re-check to be
sure.
Best regards
Volia

З павагай,
Вольга.


вт, 11 июн. 2024 г. в 20:07, rlexus90 ***@***.***>:

File: cdk-deploy/test/cdk-deploy.test.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+//   const app = new cdk.App();
+//     // WHEN
+//   const stack = new CdkDeploy.CdkDeployStack(app, 'MyTestStack');
+//     // THEN
+//   const template = Template.fromStack(stack);
+
+//   template.hasResourceProperties('AWS::SQS::Queue', {
+//     VisibilityTimeout: 300
+//   });
+});
```
Comment (bdvx) [inline cdk-deploy/test/cdk-deploy.test.ts:17]: Probably don't need that for now, especially to test SQS which is not in stack(at least yet) 🙂

File: package.json [hunk 2]
```diff
@@ -10,7 +10,9 @@
     "test:ui": "vitest --ui",
     "test:coverage": "vitest run --coverage",
     "lint": "eslint --ignore-path .gitignore --fix src",
-    "prettier": "prettier src --write"
+    "prettier": "prettier src --write",
+		"deploy": "npm run build && cd cdk-deploy && cdk deploy",
+		"destroy": "cd cdk-deploy && cdk destroy -f"
```
Comment (bdvx) [inline package.json:15]: Indentation seems to be off here

**Review by bdvx — APPROVED** [review]
Looking good overall, a few minor comments

**Pull request merged**